### ISSUE TITLE
Skip-GANomaly: Skip Connected and Adversarially Trained Encoder-Decoder Anomaly Detection

### ISSUE THREAD

**Issue by llien30** [mn1204/paper_summary#2]
# INFO

## Author
Samet Akçay, Amir Atapour-Abarghouei, Toby P. Breckon
## Conference or Year
Submitted on 25 Jan 2019
IEEE 2019
## Link
[arxiv](https://arxiv.org/abs/1901.08954)
[official GitHub](https://github.com/samet-akcay/skip-ganomaly) (Pytorch)

# Abstract
Anomaly Detection : the detection of outliers within a given data distribution 
→ most of the real-world data is highly imbalanced towards normality
→ it makes difficult to use well-established supervised learning methods

### our method

- unsupervised anomaly detection model
- trained only on the normal data
- encoder-decoder convolutional neural network with **skip connection**
   → superior reconstruction within the image and latent vector space
- success to throughly capture the multi-scale distribution of the data in high-dimensional space
- minimize the reconstruction error 
   → model can learn the distribution of normal image properly
   → high deviation between normal and abnormal
- SOTA performance in both quantitatively and qualitatively 
- made simple and effective network. easy to reproduced

# Proposed Method
### Problem Difinition
- model is trained only on normal data (in unsupervised manner)
- model is tested on both normal and abnormal data
### Dataset
<img width="758" alt="スクリーンショット 2020-02-03 11 02 41" src="https://user-images.githubusercontent.com/51707667/73620320-b5e42580-4674-11ea-9dca-3211f1de54ce.png">

## Network
<img width="747" alt="スクリーンショット 2020-02-03 11 23 53" src="https://user-images.githubusercontent.com/51707667/73620986-ad411e80-4677-11ea-88d5-8ee161d49ee3.png">
<img width="744" alt="スクリーンショット 2020-02-03 11 23 16" src="https://user-images.githubusercontent.com/51707667/73620972-96023100-4677-11ea-9207-1618535a3d30.png">
<img width="753" alt="スクリーンショット 2020-02-03 11 24 20" src="https://user-images.githubusercontent.com/51707667/73621002-bdf19480-4677-11ea-96bc-7c68eb300478.png">

### Adversarial Loss
<img width="340" alt="スクリーンショット 2020-02-03 11 39 01" src="https://user-images.githubusercontent.com/51707667/73621661-4113ea00-467a-11ea-9144-70d6f6384b4d.png">

### Contextual Loss
<img width="274" alt="スクリーンショット 2020-02-03 11 44 35" src="https://user-images.githubusercontent.com/51707667/73621765-9f40cd00-467a-11ea-9f16-99a802d5bf0f.png">

### Latent Loss
using final convolutional layer of the Discriminator
<img width="296" alt="スクリーンショット 2020-02-03 11 46 59" src="https://user-images.githubusercontent.com/51707667/73621848-f5157500-467a-11ea-9fc1-281da39fe1d8.png">

### Total Loss
<img width="522" alt="スクリーンショット 2020-02-03 11 51 30" src="https://user-images.githubusercontent.com/51707667/73621994-87b61400-467b-11ea-92a2-60ef7b4698ea.png">

## Anomaly Score
<img width="517" alt="スクリーンショット 2020-02-03 12 02 13" src="https://user-images.githubusercontent.com/51707667/73622543-0790ae00-467d-11ea-95db-76c6735e160e.png">

# Evaluation
### Datasets
- CIFER-10
       training sample →  45000
       test sample → normal : abnormal = 9000 : 6000
- University Baggage Dataset (UBA)
       abnormal : knife (63,496), gun (45,855), gun component (13,452)
       normal  : 107,472 → training : test = 80 : 20
- Full Firearm vs Operational Benign (FFOB) 
       abnormal : 4,680
       normal : 67,672
### Training Details
<img width="582" alt="スクリーンショット 2020-02-03 12 18 07" src="https://user-images.githubusercontent.com/51707667/73623124-40318700-467f-11ea-84f1-7b6f24884393.png">

## Results
<img width="753" alt="スクリーンショット 2020-02-03 12 23 37" src="https://user-images.githubusercontent.com/51707667/73623358-08770f00-4680-11ea-9c26-c1b72a133355.png">
<img width="739" alt="スクリーンショット 2020-02-03 12 33 01" src="https://user-images.githubusercontent.com/51707667/73623720-57717400-4681-11ea-8c7c-8f1eea072fbc.png">
<img width="908" alt="スクリーンショット 2020-02-03 12 57 19" src="https://user-images.githubusercontent.com/51707667/73624587-c4d2d400-4684-11ea-9cd3-a6ac42be450a.png">
<img width="444" alt="スクリーンショット 2020-02-03 12 58 16" src="https://user-images.githubusercontent.com/51707667/73624691-2d21b580-4685-11ea-8eac-52123924391d.png">
<img width="756" alt="スクリーンショット 2020-02-03 13 01 01" src="https://user-images.githubusercontent.com/51707667/73624714-3e6ac200-4685-11ea-91e8-9912071e1dc7.png">
<img width="741" alt="スクリーンショット 2020-02-03 13 04 25" src="https://user-images.githubusercontent.com/51707667/73624842-b933dd00-4685-11ea-8e52-f016b2380fe0.png">

# Contribution
achieve SOTA performance with skip-connection between networks 

# Discussion, Future Work
- to detect anomalies in higher resolution image
- to detect anomalies of the data containing temporal information 

## Date
2020/02/03
